src/azure-cli/azure/cli/command_modules/appservice/_constants.py

```diff
@@ -19,26 +18,59 @@
 NETCORE_VERSIONS = ['1.0', '1.1', '2.1', '2.2']
 DOTNET_VERSIONS = ['3.5', '4.7']
 LINUX_SKU_DEFAULT = "P1V2"
-RUNTIME_TO_DEFAULT_VERSION = {
-    'node': '8',
-    'dotnet': '2',
-    'python': '3.6',
-    'java': '8'
+HOST_VERSIONS_FUNCTIONAPP = ['2', '3']
+# host version : default node version
+NODE_VERSION_DEFAULT_FUNCTIONAPP = {
+    '2': '~10',
+    '3': '~12'
 }
-
-RUNTIME_TO_IMAGE_FUNCTIONAPP = {
-    'node': {
-        '8': 'mcr.microsoft.com/azure-functions/node:2.0-node8-appservice',
-        '10': 'mcr.microsoft.com/azure-functions/node:2.0-node10-appservice'
-    },
-    'python': {
-        '3.6': 'mcr.microsoft.com/azure-functions/python:2.0-python3.6-appservice',
-        '3.7': 'mcr.microsoft.com/azure-functions/python:2.0-python3.7-appservice'
+# host version -> runtime : default runtime version
+RUNTIME_TO_DEFAULT_VERSION_FUNCTIONAPP = {
+    '2': {
+        'node': '8',
+        'dotnet': '2',
+        'python': '3.6',
+        'java': '8'
     },
-    'dotnet': {
-        '2': 'mcr.microsoft.com/azure-functions/dotnet:2.0-appservice'
+    '3': {
+        'node': '12',
+        'dotnet': '3',
+        'python': '3.6',
+        'java': '8'
+    }
+}
+# host version -> runtime -> runtime version : container image
+RUNTIME_TO_IMAGE_FUNCTIONAPP = {
+    '2': {
+        'node': {
+            '8': 'mcr.microsoft.com/azure-functions/node:2.0-node8-appservice',
+            '10': 'mcr.microsoft.com/azure-functions/node:2.0-node10-appservice'
+        },
+        'python': {
+            '3.6': 'mcr.microsoft.com/azure-functions/python:2.0-python3.6-appservice',
+            '3.7': 'mcr.microsoft.com/azure-functions/python:2.0-python3.7-appservice'
+        },
+        'dotnet': {
+            '2': 'mcr.microsoft.com/azure-functions/dotnet:2.0-appservice'
+        },
+        'java': {
+            '8': 'mcr.microsoft.com/azure-functions/java:2.0-java8-appservice'
+        }
     },
-    'java': {
-        '8': 'mcr.microsoft.com/azure-functions/java:2.0-java8-appservice'
+    '3': {
+        'node': {
+            '10': 'mcr.microsoft.com/azure-functions/node:3.0-node10-appservice',
+            '12': 'mcr.microsoft.com/azure-functions/node:3.0-node12-appservice'
+        },
+        'python': {
+            '3.6': 'mcr.microsoft.com/azure-functions/python:3.0-python3.6-appservice',
+            '3.7': 'mcr.microsoft.com/azure-functions/python:3.0-python3.7-appservice'
```

Comment:
Could you help me add 3.8 here?
`'3.8': 'mcr.microsoft.com/azure-functions/python:3.0-python3.8-appservice'`

Reply:
Done -- changed the defaults as well per offline conversation